lib/icingadb/icingadb-worker.cpp

```diff
@@ -0,0 +1,531 @@
+/* Icinga 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+#include "icingadb/icingadb.hpp"
+#include "base/logger.hpp"
+#include <vector>
+
+using namespace icinga;
+
+/**
+ * Generate a unique queue ID for the given config object [and dependency group].
+ *
+ * This function generates a unique string identifier for a pending item based on the provided object pointer
+ * and an optional dependency group pointer. The unique ID is created by combining the hash values of the config
+ * object and dependency group (if provided) to ensure uniqueness. The resulting ID is prefixed with "DUMMY_" to
+ * indicate that it is a generated identifier intended for use in the pending items queue but never used as an
+ * actual Redis hash key.
+ *
+ * @param obj The configuration object for which the unique ID is being generated.
+ * @param depGroup The dependency group associated with the pending item (optional).
+ *
+ * @return A unique string identifier for the pending item.
+ */
+static String MakeUniqueQueueID(const ConfigObject::Ptr& obj = nullptr, const DependencyGroup::Ptr& depGroup = nullptr)
+{
+	size_t seed = 0;
+	if (obj) {
+		boost::hash_combine(seed, obj);
+	}
+	if (depGroup) {
+		boost::hash_combine(seed, depGroup);
+	}
+	return "DUMMY_" + std::to_string(seed);
+}
```

Comment:
All the hashing in C++ based on `size_t` isn't really meant to prevent collisions. Sure, collisions should be unlikely, but code using it must assume there are collisions. Keeping the overall function signature, I would rather just write both pointer values to the string, that's guaranteed that there won't be any collisions then.

However, I'm also question why this function even needs to exist, i.e. why you're mapping this to a string. Couldn't you also use some `std::variant` containing the different pointer types as the ID (or lookup key) instead?